src/Chainweb/Pact/PactService.hs

```diff
@@ -445,15 +441,32 @@ validateChainwebTxsPreBlock dbEnv cp blockOriginationTime bh hash txs = do
           let validations = (const (b >= limitInCoin)) : validators
           return $! all ($ tx) validations
 
-    checkOne tx = do
+    -- skip validation for genesis -- gas accounts / etc don't exist yet
+    checkOne tx = if bh == 0 then return True else checkOneReal tx
+    checkOneReal tx = do
         let pactHash = view P.cmdHash tx
         mb <- _cpLookupProcessedTx cp pactHash
         if mb == Nothing
-        -- prop_tx_ttl_newblock_validate
-        then checkAccount [k] tx
-        else return False
+            -- prop_tx_ttl_newblock_validate
+            then checkAccount [checkTimes] tx
+            else return False
       where
-        k tx' = bh /= 0 && timingsCheck blockOriginationTime (payloadObj <$> tx')
+        checkTimes tx' = timingsCheck blockOriginationTime (payloadObj <$> tx')
+
+
+validateChainwebTxsPreBlock
+    :: PactDbEnv'
+    -> Checkpointer
+    -> BlockCreationTime
+    -> BlockHeight
+    -> BlockHash
+    -> Vector ChainwebTransaction
+    -> IO (Vector Bool)
+validateChainwebTxsPreBlock dbEnv cp blockOriginationTime bh hash txs = do
+    lb <- _cpGetLatestBlock cp
+    when (Just (pred bh, hash) /= lb) $
+        fail "internal error: restore point is wrong, refusing to validate."
```

Comment:
Instead of this being a general `fail`, why don't we make this a `PactInternalError`?

Reply:
done